### ISSUE TITLE
TS breaks with type alias as index signature

### ISSUE THREAD

**Issue by theseyi** [Microsoft/TypeScript#19146]
**TypeScript Version:**  2.6.0-dev.201xxxxx

**Code**

```ts
enum StringEnum {
  Name = 'NAME',
  Email = 'EMAIL'
}

type EnumKeyed  = {
  [K in StringEnum]: string;
}

const obj: EnumKeyed = {
  NAME: 'aValue',
  EMAIL: 'bValue'
};

const array = Object.keys(obj) as Array<StringEnum>;
const result = array.map((key: StringEnum) => obj[key]);
```

**Expected behavior:**
`result // ['aValue', 'bValue'];`
**Actual behavior:**

TS throws:
```
Element implicitly has an 'any' type because type 'EnumKeyed' has no index signature.
const obj: EnumKeyed
```

**Comment by ahejlsberg** [comment]
This looks like a duplicate of #16760 which is already fixed in 2.6. It doesn't reproduce with `typescript@next`. Are you sure you're on version 2.6?

**Comment by theseyi** [comment]
Ah, apologies, was running on typescript@latest. This is fixed in @2.6.0-dev.20171013. Thanks so much @ahejlsberg
